docs/dev/guidelines/server.rst

```diff
@@ -459,19 +451,69 @@ The table contains all annotations for the corresponding minimum role including
 | ADMIN            | @EnforceAdmin                          | /api/admin/     | web.rest.admin |
 +------------------+----------------------------------------+-----------------+----------------+
-| INSTRUCTOR       | @EnforceAtLeastInstructor              | /api/           | web.rest       |
+| INSTRUCTOR       | @EnforceAtLeastInstructorInResource    | /api/           | web.rest       |
 +------------------+----------------------------------------+-----------------+----------------+
-| EDITOR           | @EnforceAtLeastEditor                  | /api/           | web.rest       |
+| EDITOR           | @EnforceAtLeastEditorInResource        | /api/           | web.rest       |
 +------------------+----------------------------------------+-----------------+----------------+
-| TA               | @EnforceAtLeastTutor                   | /api/           | web.rest       |
+| TA               | @EnforceAtLeastTutorInResource         | /api/           | web.rest       |
 +------------------+----------------------------------------+-----------------+----------------+
-| USER             | @EnforceAtLeastStudent                 | /api/           | web.rest       |
+| USER             | @EnforceAtLeastStudentInResource       | /api/           | web.rest       |
 +------------------+----------------------------------------+-----------------+----------------+
 | ANONYMOUS        | @EnforceNothing                        | /api/public/    | web.rest.open  |
 +------------------+----------------------------------------+-----------------+----------------+
 
 If, for some reason, you need to deviate from these rules, use ``@ManualConfig``. Use this annotation only if absolutely necessary as it will exclude the endpoint from the automatic authorization tests.
 
+Implicit pre- and post-authorization annotations
+------------------------------------------------
+
+The following example makes the call only accessible to ADMIN and INSTRUCTOR users and then checks the access rights to the course in the database:
+
+Do **not** write
+
+.. code-block:: java
+
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Void> enableLearningPathsForCourse(@PathVariable long courseId) {
+        var course = courseRepository.findById(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        [...]
+        return ResponseEntity.ok().build();
+    }
+
+Instead, use the following annotation:
+
+.. code-block:: java
+
+    @EnforceAtLeastInstructorInCourse
+    public ResponseEntity<Void> enableLearningPathsForCourse(@PathVariable long courseId) {
+        [...]
+        return ResponseEntity.ok().build();
+    }
+
+Utilizing the ``EnforceRoleInResource`` annotation reduces the amount of code and makes the code more readable and maintainable. Also it is faster than the explicit authorization check since we don't have to load the user and course into JAVA as the check is performed directly in a database query.
```

Comment:
```suggestion
Utilizing the ``EnforceRoleInResource`` annotation reduces the amount of code and makes the code more readable and maintainable. Also it is faster than the explicit authorization check since we don't have to load the user and course into Java as the check is performed directly in a database query.
```